### ISSUE TITLE
Qwen3.5-2B BM1688 seq2048 dynamic prefill degrades around 516 tokens

### ISSUE THREAD

**Issue by tianna1121** [sophgo/LLM-TPU#154]
## Summary

We are testing the official Qwen3.5-2B BM1688 dynamic bmodel and observe a very sharp prefill degradation boundary around 512 tokens.

The model works correctly when the prompt is about 510 tokens, but starts to produce incorrect/garbage output or immediately stops when the prompt is about 516 tokens. This is unexpected because the bmodel metadata and official documentation indicate `max_input_length=1024` and `seq_length=2048`.

## Model / bmodel

BModel file from the official Qwen3_5 BM1688 package:

```text
qwen3.5-2b-int4-autoround_w4bf16_seq2048_bm1688_2core_dynamic_20260415_212627.bmodel
```

Relevant official doc page:

```text
models/Qwen3_5/README.md
```

The documented compile/run configuration appears to be:

```text
--max_input_length 1024
-s 2048
BM1688 2core dynamic
```

## Runtime environment

Hardware / runtime:

```text
Device: BM1688
OS: SylixOS
Runtime path: BMRT / bmlib C++ service, aligned with LLM-TPU Qwen3_5 cpp_demo_pp block scheduling logic
History Support: False
Service reports: max_input=1024, seqlen=2048
```

We currently do not have a BM1688 Linux board available for comparison, so this report is based on BM1688 SylixOS + Sophon BMRT/bmlib runtime.

## Reproduction behavior

We tested prompts with controlled token lengths and asked the model to retrieve a simple key/value from the prompt. The behavior changes sharply between 510 and 516 prompt tokens:

```text
prompt_tokens=498 -> correct: code=B-150, area=west, threshold=73
prompt_tokens=504 -> correct: code=C-152, area=west, threshold=73
prompt_tokens=510 -> correct: code=C-154, area=west, threshold=73
prompt_tokens=516 -> wrong: only outputs "user"
prompt_tokens=522/528/534/540/546/552/558/564 -> wrong: mostly outputs only one unrelated token
prompt_tokens=588/678/768/858/948 -> still wrong, usually only 1 token
prompt_tokens>1024 -> rejected by our service guard, service remains healthy
```

Short prompt + long generation works normally, so this does not look like an output length issue:

```text
prompt_tokens=26
requested max_tokens=320
actual completion_tokens=320
service health after generation: ok
```

We also raised frontend/API output limits up to 1000 tokens and confirmed that short prompts can generate 1000 tokens successfully. The failure is tied to prefill input length, not generation length.

## BModel inspection

After stopping the online service and releasing NPU memory, we inspected the bmodel successfully. The key shapes look consistent with `max_input_length=1024` / `seq_length=2048`:

```text
network_count=52
embedding: input=[1,1024], output=[1,1024,2048]
non-FA block examples: dynamic, input=[1,1024,2048], state input=[1,16,128,128]
FA block examples: dynamic, position_ids=[3,1024], attention_mask=[1,1,1024,1024]
block_cache_* decode: SEQLEN=2048
lm_head: input=[1,2048]
History Support: False
```

This makes it look like the metadata supports 1024-token prefill, but actual inference quality collapses just after ~512 tokens.

## Things we already checked

1. Not an HTTP/frontend truncation issue: backend sees `max_input=1024` and accepts prompts above 512.
2. Not a max output token issue: short prompts can generate 320/1000 tokens correctly.
3. Not obviously a dynamic shape argument bug in our wrapper: we tried a probe build that forces prefill to use fixed 1024 padding instead of the dynamic `token_length` shape. The result did not improve; 518/638/838-token prompts still failed, only first-token latency became fixed and slower.
4. Our SylixOS `block.cpp` scheduling logic was compared with the official `LLM-TPU/models/Qwen3_5/cpp_demo_pp/block.cpp`; no meaningful deviation was found in the layer/block invocation logic.

## Question

Is there a known limitation or bug for Qwen3.5-2B BM1688 `seq2048 dynamic` bmodels around 512-token prefill length?

Specifically, should this official bmodel be expected to handle prompts between 512 and 1024 tokens correctly on BM1688, or is there an export/runtime constraint related to Qwen3.5 linear-attention / recurrent state / attention mask handling that effectively limits usable prefill length to 512?

Any suggestions for additional diagnostics would be helpful. If needed, we can provide more detailed logs from the BM1688 SylixOS runtime and the controlled prompt test cases.

**Comment by HarmonyHu** [comment]
@tianna1121 Thank you for your issue report. I will check it.

**Comment by HarmonyHu** [comment]
it has been fixed by latest commit: https://github.com/sophgo/LLM-TPU/commit/63d8ab34dab5fee381f3f492bf86c061b862ffa6, you can have a try. @tianna1121
